lib/stagger/client.rb

```diff
@@ -21,6 +22,23 @@ def shutdown
       @conn.shutdown
     end
 
+    # Send process information like the name to the client
+    def register_process(tags={})
+      return if tags.empty?
+
+      @process_info = tags.each_with_object({}) do |(k, v), hash|
+        hash[to_utf8(k)] = to_utf8(v)
+      end
+
+      send_register_process
+    end
```

Comment:
Please can this method either:
 - be a regular setter for `@process_info` please, and not make a follow on call to `send_register_process`, that rather confuses things.
 - take the arguments, clean them and send them on the connection without memoising them.

I don't even think it's correct in the one case that it's called currently (line 15). `setup_connection` has already been called on line 11 and will call `send_register_process` when the connection completes.

The call to `send_register_process` on line 33 will always be a silent no-op because `send_register_process` returns immediately if called when `!@connected`. That should be an error, IMO.